File: src/ZIG/hello.zig
// zig: zig run hello.zig
// zig (build): zig build-exe hello.zig && ./hello
// zig (release): zig build-exe -O ReleaseFast hello.zig && ./hello
// zig (cross): zig build-exe -target x86_64-windows hello.zig && wine hello.exe
// zig (test): zig test hello.zig
// zig (check): zig build-exe --check hello.zig
// zig (build system): zig build
const std = @import("std");

pub fn main() !void {
    std.debug.print("Hello from Zig!\n", .{});
}

File: src/RUST/hello.rs
// hello.rs
// rustc: rustc hello.rs && ./hello
// cargo: cargo run (with Cargo.toml)
// cargo (build): cargo build && ./target/debug/hello
// cargo (release): cargo build --release && ./target/release/hello
// rustc (release): rustc -O hello.rs && ./hello
// cross: cross build --target x86_64-pc-windows-gnu && wine target/x86_64-pc-windows-gnu/debug/hello.exe
// miri: cargo +nightly miri run
fn main() {
    println!("Hello from Rust!");
}
